sphinx/ext/javadoctest.py

```diff
@@ -0,0 +1,118 @@
+"""Mimic javadoctest in Sphinx.
+
+The extension automatically execute code snippets and checks their results on Java engine.
+"""
+
+import subprocess
+from typing import Any, Dict
+
+from sphinx.application import Sphinx
+from sphinx.ext.doctest import (
+    DocTestBuilder,
+    TestcodeDirective,
+    TestoutputDirective,
+    doctest,
+    sphinx,
+)
+from sphinx.locale import __
+
+
+class JavaTestcodeDirective(TestcodeDirective):
+    def run(self):
+        node_list = super().run()
+        print("check_node_list")
```

Comment:
I know that this is a draft, but remember to remove the `print` (maybe by adding some todo comment).

Reply:
Deleted